### ISSUE TITLE
Adding a `topn` parameter to the Exhaustive Feature Selector

### ISSUE THREAD

**Issue by rasbt** [rasbt/mlxtend#610]
When running the `ExhaustiveFeatureSelector`, it is not unusual that an extraordinary large number of results get collected (due to the many combinations to evaluate). Some users reported that this leads to memory issues when calling the `get_metric_dict()` method that constructs a DataFrame containing the results.

Thus, it would be good to add a parameter like "topn" for the `get_metric_dict()` method that only returns the top n results. 

Also, there seems to be an inconsistency that `.subsets_` for efs strats at key 0 whereas `sfs starts at 1` which should ideally be fixed.

This issue is based on the discussion on the mailinglist at https://groups.google.com/forum/#!topic/mlxtend/2h7WNBn_A38

**Comment by ScorpionXiezi** [comment]
I will be working on this issue. Follow up will be posted this week.

**Comment by rasbt** [comment]
Sounds good. Please note that I have to enter the grades on Monday, so please don't wait too long.

**Comment by sagnik1511** [comment]
Is this issue still alive? I'll be happy to contribute.

**Comment by rasbt** [comment]
Yes, it's still alive! And I would definitely welcome contributions! 🙌
